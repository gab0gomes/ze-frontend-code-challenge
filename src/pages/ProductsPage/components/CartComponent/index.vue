<template>
  <div class="cart-container">
    <img :src="cart" alt="uma figura de um carrinho de compras" />
    <p>
      {{ getFormattedTotalPrice }}
    </p>
  </div>
</template>

<script>
import { mapGetters } from 'vuex'
import cart from '../../../../../static/images/cart.png'

export default {
  data() {
    return {
      cart
    }
  },
  computed: {
    ...mapGetters('cart', ['getFormattedTotalPrice'])
  }
}
</script>

<style scoped>
.cart-container {
  border-radius: 8px;
  height: 48px;
  max-width: 128px;
  border: 1px solid #3c2946;
  position: fixed;
  bottom: 32px;
  right: 16px;
  padding: 8px;
  display: flex;
  justify-content: center;
  background-color: #ffd571;
  align-items: center;
  color: #3c2946;
  box-shadow: 4px 4px #3c2946;
  z-index: 999;
}

.cart-container img {
  max-width: 24px;
}

.cart-container p {
  margin: 0;
  margin-left: 8px;
}
</style>
