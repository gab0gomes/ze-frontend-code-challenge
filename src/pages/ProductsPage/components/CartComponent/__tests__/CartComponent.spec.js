import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'

import CartComponent from '../index'

import cartStore from '@/core/store/cart'

const localVue = createLocalVue()
localVue.use(Vuex)

const store = () =>
  new Vuex.Store({
    modules: {
      cart: {
        ...cartStore
      }
    }
  })

describe('CartComponent', () => {
  it('render correctly', async () => {
    const wrapper = shallowMount(CartComponent, {
      localVue,
      store: store()
    })

    expect(wrapper.element).toMatchSnapshot()

    wrapper.destroy()
  })
})
