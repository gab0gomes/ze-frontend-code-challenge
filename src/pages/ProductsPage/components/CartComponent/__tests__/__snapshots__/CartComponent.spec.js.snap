// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartComponent render correctly 1`] = `
<div
  class="cart-container"
>
  <img
    alt="uma figura de um carrinho de compras"
    src="static/images/cart.png"
  />
   
  <p>
    
    R$Â 0,00
  
  </p>
</div>
`;
